### PR TITLE
Feature/add schema class

### PR DESCRIPTION
**Pull request by dvd9604**
tests coming soonish, maybe tomorrow. Still thinking about best way to store complex reoccurrence.

### CODE REVIEW

File: Serverless/AWS/API/querydb.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import json
```
Comment (bsneider) [inline Serverless/AWS/API/querydb.py:1]: Checkout the LambdaResearch repo I had updated this class to include cors headers in response

File: Serverless/AWS/DynamoDBGeo/fix_dicts.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+import datetime
+
+mydict = {
+    'hash': 'hello',
+    'number_int': 10,
+    'number_float': 12.23
+}
+
+def get_type_string(item):
```
Comment (bsneider) [inline Serverless/AWS/DynamoDBGeo/fix_dicts.py:10]: this stuff is not needed after I fixed the dynamodb-geo package see the Lambda code and my fork of dynamodb-geo

File: Serverless/AWS/schema/schema.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+     tag=EventCategory.game,
+     name="my pickup game",
+     description="fun game in SA",
+     location = (72,84),
+     organizer="dvd9604",
+     email="dvd9604@gmail.com",
+     phone="8008888888",
+     website="http://www.github.com/dvd9604",
+     start_time = datetime(2021,1,1,4,30).timestamp(),
+     repeat_time = (timedelta(days=7).total_seconds(), timedelta(hours=1).total_seconds()),
```
Comment (bsneider) [inline Serverless/AWS/schema/schema.py:84]: cool

File: Serverless/AWS/schema/schema.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+
+
+if __name__ == "__main__":
+    
+    e = Event(
+     src=EventSource.ultimate_central,
+     tag=EventCategory.game,
+     name="my pickup game",
+     description="fun game in SA",
```
Comment (bsneider) [inline Serverless/AWS/schema/schema.py:77]: lol nice

File: Serverless/AWS/schema/schema.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        self.name = kwargs.get('name')
+        self.description =  kwargs.get('description', None)
+        self.location =  kwargs.get('location', None)
+        self.contact = {
+            "organizer": kwargs.get('organizer', None),
+            "email":  kwargs.get('email', None),
+            "phone":  kwargs.get('phone', None),
+            "website":  kwargs.get('website', None)
+        }
+        self.schedule = {
```
Comment (bsneider) [inline Serverless/AWS/schema/schema.py:27]: I've been thinking, should we have multiple schedules, that way we can do more than one day in an irregular repeating manner in one request, or is it best to make it each on a separate event?

File: Serverless/AWS/schema/schema.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    social = 's'
+    league = 'l'
+
+class EventSource(Enum):
+    pickup_ultimate = "pu"
+    ultimate_central = "uc"
+    
+class Event():
+    def __init__(self, src, tag, **kwargs):
+        self._id = self.create_id(src, kwargs.get('src_id', None))
```
Comment (bsneider) [inline Serverless/AWS/schema/schema.py:17]: _ underscore is extra space

File: Serverless/AWS/schema/schema.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    game = 'g'
+    tournament = 't'
+    social = 's'
+    league = 'l'
+
+class EventSource(Enum):
+    pickup_ultimate = "pu"
+    ultimate_central = "uc"
+    
+class Event():
```
Comment (bsneider) [inline Serverless/AWS/schema/schema.py:15]: if we make this call inherit __dict__ we can show in the dict constructor the shorter method names

Comment (bsneider) [reply Serverless/AWS/schema/schema.py:15]: actually that won't work for random fields in metadata, so I like your approach

File: Serverless/AWS/API/querydb.py (hunk 1 above)
Comment (bsneider) [reply Serverless/AWS/API/querydb.py:1]: https://github.com/bsneider/LambdaAPIResearch/tree/main/ansible/pickup

Comment (dvd9604) [reply Serverless/AWS/API/querydb.py:1]: Cool, should I be pushing to that repo?

File: Serverless/AWS/DynamoDBGeo/fix_dicts.py (hunk 2 above)
Comment (dvd9604) [reply Serverless/AWS/DynamoDBGeo/fix_dicts.py:10]: nice that typing stuff was annoying af

File: Serverless/AWS/schema/schema.py (hunk 5 above)
Comment (dvd9604) [reply Serverless/AWS/schema/schema.py:33]: I decided to copy the other guys and encode like so `(day of week, start time, end time)`. `day of week` is encoded as a digit.

File: Serverless/AWS/schema/schema.py (hunk 6 above)
Comment (dvd9604) [reply Serverless/AWS/schema/schema.py:22]: When minified it becomes just `_`

**Comment by dvd9604** [comment]
AWS was giving me memory and permissions issues running the func. I do think it will work though. I'll work on it some more tomorrow.

**Pull request merged**